app/controllers/data_types_controller.rb

```diff
@@ -4,17 +4,29 @@ def index
   end
 
   def new
+    @data_types = DataType.all
     @data_type = DataType.new
   end
 
   def edit
+    @data_types = DataType.all
     @data_type = DataType.find(params[:id])
   end
 
   def create
     @data_type = DataType.new(data_type_params)
-
+    @years = Year.all
     @data_type.save
+
+    @years.each do |year|
+      @data_value = DataValue.new(value: 0.0, year_id: year.id, data_type_id: @data_type.id)
+      if @data_value.save
+        year.data_values << @data_value
+      else
+        render 'new'
+      end
+    end
+
     redirect_to action: "index"
```

Comment:
Are there potential problems with possibly to hit both `render` and `redirect_to`? Not sure if Rails will complain about this.